blog/views.py

```diff
@@ -1,25 +1,55 @@
 from django.contrib import messages
-from django.shortcuts import get_object_or_404, render, redirect
+from django.shortcuts import get_object_or_404, render, redirect, HttpResponseRedirect
 
 from .models import Blog
 from .forms import BlogForm
 
 
 def blog_list(request):
-    pass
+    blog_list = Blog.objects.all()
+    context = {'blog_list': blog_list}
+    return render(request, 'blog/list.html', context)
 
 
-def blog_detail(request):
-    pass
+def blog_detail(request, pk):
+    context = {}
+    context["data"] = Blog.objects.get(id=pk)
```

Comment:
sorry going in reverse, see my comment below about `get_object_or_404`, which prevents having to do error handling, in this case doing a `get` on a non existing id Django would raise a `DoesNotExist` exception you would have to catch